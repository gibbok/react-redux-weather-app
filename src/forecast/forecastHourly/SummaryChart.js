import React from 'react'
import { LineChart, Line, CartesianGrid, XAxis, YAxis, Tooltip } from 'recharts'
import SummaryChartIcon from './SummaryChartIcon'
import moment from 'moment'

const SummaryChart = ({ forecastHourly }) => {
  let forecastHourlyFormat = forecastHourly.map(data => {
    return {
      momentId: data.momentId,
      timespan: data.timespan,
      temp: Math.round(data.temp),
      weatherIconCode: data.weatherIconCode,
      rain: data.rain,
      hour: moment.unix(data.timespan).format('hh A')
    }
  })
  return (
    <LineChart width={800} height={500} data={forecastHourlyFormat} margin={{ top: 20, right: 20, bottom: 20, left: 20 }}>
      <Line type='monotone' dataKey='temp' stroke='#8884d8' dot={<SummaryChartIcon />} />
      <CartesianGrid stroke='#ccc' strokeDasharray='5 5' />
      <XAxis dataKey='hour' />
      <YAxis dataKey='temp' />
      <Tooltip />
    </LineChart>
  )
}

// const SummaryChart = ({ forecastHourly }) => (
//   <LineChart width={800} height={500} data={forecastHourly} margin={{ top: 20, right: 20, bottom: 20, left: 20 }}>
//     <Line type='monotone' dataKey='temp' stroke='#8884d8' dot={<SummaryChartIcon />} />
//     <CartesianGrid stroke='#ccc' strokeDasharray='5 5' />
//     <XAxis dataKey='hour' />
//     <Tooltip />
//   </LineChart>
// )

export default SummaryChart
