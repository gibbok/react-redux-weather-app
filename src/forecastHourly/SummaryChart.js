import React from 'react'
import { LineChart, Line, CartesianGrid, XAxis, Tooltip } from 'recharts'
import SummaryChartIcon from './SummaryChartIcon'

const SummaryChart = ({ forecastHourly }) => (
  <LineChart width={500} height={250} data={forecastHourly} margin={{ top: 20, right: 20, bottom: 20, left: 20 }}>
    <Line type='monotone' dataKey='temp' stroke='#8884d8' dot={<SummaryChartIcon />} />
    <CartesianGrid stroke='#ccc' strokeDasharray='5 5' />
    <XAxis dataKey='hour' />
    <Tooltip />
  </LineChart>
)

export default SummaryChart
