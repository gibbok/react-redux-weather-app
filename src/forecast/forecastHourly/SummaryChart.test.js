// @flow
import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import SummaryChart from './SummaryChart'

describe('<SummaryChart />', () => {
  it('should render', () => {
    const data = [
      {
        momentId: '2017:8:12:2',
        timespan: 1505217600,
        temp: 15.52,
        weatherIconCode: 804,
        weatherDescription: 'Clouds',
        windDegree: 344,
        windSpeed: 1.3,
        humidityUnit: '%',
        humidityValue: 89},
      {
        momentId: '2017:8:12:2',
        timespan: 1505228400,
        temp: 21.77,
        weatherIconCode: 801,
        weatherDescription: 'Clouds',
        windDegree: 266.501,
        windSpeed: 1.27,
        humidityUnit: '%',
        humidityValue: 67
      }
    ]
    const wrapper = shallow(
      <SummaryChart
        forecastHourly={data}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
/*
[{"momentId":"2017:8:12:2","timespan":1505217600,"temp":15.52,"weatherIconCode":804,"weatherDescription":"Clouds","windDegree":344,"windSpeed":1.3,"humidityUnit":"%","humidityValue":89},{"momentId":"2017:8:12:2","timespan":1505228400,"temp":21.77,"weatherIconCode":801,"weatherDescription":"Clouds","windDegree":266.501,"windSpeed":1.27,"humidityUnit":"%","humidityValue":67},{"momentId":"2017:8:12:2","timespan":1505239200,"temp":24.13,"weatherIconCode":803,"weatherDescription":"Clouds","windDegree":266.502,"windSpeed":1.42,"humidityUnit":"%","humidityValue":60},{"momentId":"2017:8:12:2","timespan":1505250000,"temp":24.61,"weatherIconCode":803,"weatherDescription":"Clouds","windDegree":237.503,"windSpeed":1.96,"humidityUnit":"%","humidityValue":56},{"momentId":"2017:8:13:3","timespan":1505260800,"temp":21.57,"weatherIconCode":803,"weatherDescription":"Clouds","windDegree":227.007,"windSpeed":2.56,"humidityUnit":"%","humidityValue":63},{"momentId":"2017:8:13:3","timespan":1505271600,"temp":19.27,"weatherIconCode":803,"weatherDescription":"Clouds","windDegree":258.002,"windSpeed":3.41,"humidityUnit":"%","humidityValue":77},{"momentId":"2017:8:13:3","timespan":1505282400,"temp":17.23,"weatherIconCode":802,"weatherDescription":"Clouds","windDegree":254.503,"windSpeed":2.87,"humidityUnit":"%","humidityValue":90},{"momentId":"2017:8:13:3","timespan":1505293200,"temp":16.09,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":268.001,"windSpeed":1.46,"humidityUnit":"%","humidityValue":95},{"momentId":"2017:8:13:3","timespan":1505304000,"temp":17.86,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":256.001,"windSpeed":1.11,"humidityUnit":"%","humidityValue":88},{"momentId":"2017:8:13:3","timespan":1505314800,"temp":24.27,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":242.001,"windSpeed":1.21,"humidityUnit":"%","humidityValue":67},{"momentId":"2017:8:13:3","timespan":1505325600,"temp":25.44,"weatherIconCode":802,"weatherDescription":"Clouds","windDegree":239.5,"windSpeed":1.78,"humidityUnit":"%","humidityValue":60},{"momentId":"2017:8:13:3","timespan":1505336400,"temp":25.49,"weatherIconCode":802,"weatherDescription":"Clouds","windDegree":216.001,"windSpeed":2.71,"humidityUnit":"%","humidityValue":57},{"momentId":"2017:8:14:4","timespan":1505347200,"temp":23.41,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":211.003,"windSpeed":3.61,"humidityUnit":"%","humidityValue":64},{"momentId":"2017:8:14:4","timespan":1505358000,"temp":21.22,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":211.503,"windSpeed":2.51,"humidityUnit":"%","humidityValue":81,"rain":0.08},{"momentId":"2017:8:14:4","timespan":1505368800,"temp":20.15,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":197.504,"windSpeed":1.06,"humidityUnit":"%","humidityValue":91,"rain":0.12},{"momentId":"2017:8:14:4","timespan":1505379600,"temp":19.78,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":87.5018,"windSpeed":1.26,"humidityUnit":"%","humidityValue":96,"rain":0.04},{"momentId":"2017:8:14:4","timespan":1505390400,"temp":20.39,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":90.0029,"windSpeed":1.42,"humidityUnit":"%","humidityValue":95,"rain":0.19},{"momentId":"2017:8:14:4","timespan":1505401200,"temp":22.76,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":256.5,"windSpeed":1.16,"humidityUnit":"%","humidityValue":86,"rain":0.15},{"momentId":"2017:8:14:4","timespan":1505412000,"temp":24.57,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":275.503,"windSpeed":1.2,"humidityUnit":"%","humidityValue":75,"rain":0.08},{"momentId":"2017:8:14:4","timespan":1505422800,"temp":23.42,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":201.503,"windSpeed":1.55,"humidityUnit":"%","humidityValue":80,"rain":0.9},{"momentId":"2017:8:15:5","timespan":1505433600,"temp":21.61,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":144.502,"windSpeed":0.97,"humidityUnit":"%","humidityValue":87,"rain":0.36},{"momentId":"2017:8:15:5","timespan":1505444400,"temp":20.45,"weatherIconCode":803,"weatherDescription":"Clouds","windDegree":179,"windSpeed":2.17,"humidityUnit":"%","humidityValue":93},{"momentId":"2017:8:15:5","timespan":1505455200,"temp":20.06,"weatherIconCode":803,"weatherDescription":"Clouds","windDegree":279.501,"windSpeed":1.06,"humidityUnit":"%","humidityValue":96},{"momentId":"2017:8:15:5","timespan":1505466000,"temp":19.7,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":341.009,"windSpeed":0.96,"humidityUnit":"%","humidityValue":97,"rain":0.02},{"momentId":"2017:8:15:5","timespan":1505476800,"temp":20.72,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":84.0001,"windSpeed":1.61,"humidityUnit":"%","humidityValue":94,"rain":0.01},{"momentId":"2017:8:15:5","timespan":1505487600,"temp":24.35,"weatherIconCode":801,"weatherDescription":"Clouds","windDegree":110,"windSpeed":1.51,"humidityUnit":"%","humidityValue":79},{"momentId":"2017:8:15:5","timespan":1505498400,"temp":22.8,"weatherIconCode":501,"weatherDescription":"Rain","windDegree":142.501,"windSpeed":1.6,"humidityUnit":"%","humidityValue":93,"rain":3.12},{"momentId":"2017:8:15:5","timespan":1505509200,"temp":23.85,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":128.503,"windSpeed":1.52,"humidityUnit":"%","humidityValue":88,"rain":0.029999999999999},{"momentId":"2017:8:16:6","timespan":1505520000,"temp":21.52,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":141.502,"windSpeed":1.37,"humidityUnit":"%","humidityValue":88},{"momentId":"2017:8:16:6","timespan":1505530800,"temp":19.29,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":144.503,"windSpeed":1.11,"humidityUnit":"%","humidityValue":96},{"momentId":"2017:8:16:6","timespan":1505541600,"temp":18.16,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":83.5063,"windSpeed":1.16,"humidityUnit":"%","humidityValue":98},{"momentId":"2017:8:16:6","timespan":1505552400,"temp":17.5,"weatherIconCode":801,"weatherDescription":"Clouds","windDegree":66.5119,"windSpeed":1.42,"humidityUnit":"%","humidityValue":99},{"momentId":"2017:8:16:6","timespan":1505563200,"temp":18.96,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":63.5012,"windSpeed":2.06,"humidityUnit":"%","humidityValue":100,"rain":0.07},{"momentId":"2017:8:16:6","timespan":1505574000,"temp":21.84,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":71.0016,"windSpeed":1.97,"humidityUnit":"%","humidityValue":89,"rain":0.09},{"momentId":"2017:8:16:6","timespan":1505584800,"temp":24.75,"weatherIconCode":500,"weatherDescription":"Rain","windDegree":66.0077,"windSpeed":2.21,"humidityUnit":"%","humidityValue":75,"rain":0.0025000000000004},{"momentId":"2017:8:16:6","timespan":1505595600,"temp":24.83,"weatherIconCode":802,"weatherDescription":"Clouds","windDegree":86.5009,"windSpeed":2.61,"humidityUnit":"%","humidityValue":70},{"momentId":"2017:8:17:0","timespan":1505606400,"temp":22.45,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":85.5054,"windSpeed":2.21,"humidityUnit":"%","humidityValue":76},{"momentId":"2017:8:17:0","timespan":1505617200,"temp":19.45,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":69.5024,"windSpeed":2.1,"humidityUnit":"%","humidityValue":91},{"momentId":"2017:8:17:0","timespan":1505628000,"temp":17.56,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":49.5046,"windSpeed":1.29,"humidityUnit":"%","humidityValue":99},{"momentId":"2017:8:17:0","timespan":1505638800,"temp":16.75,"weatherIconCode":800,"weatherDescription":"Clear","windDegree":356.002,"windSpeed":1.42,"humidityUnit":"%","humidityValue":99}]
*/
