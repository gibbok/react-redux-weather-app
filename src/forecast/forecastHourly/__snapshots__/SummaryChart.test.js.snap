// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryChart /> should render 1`] = `
<AnimationDecorator(ComposedDataDecorator(LineChart))
  data={
    Array [
      Object {
        "day": "Tue, Sep 12",
        "hour": "02 PM",
        "momentId": "2017:8:12:2",
        "rain": undefined,
        "temp": 16,
        "timespan": 1505217600,
        "weatherIconCode": 804,
      },
      Object {
        "day": "Tue, Sep 12",
        "hour": "05 PM",
        "momentId": "2017:8:12:2",
        "rain": undefined,
        "temp": 22,
        "timespan": 1505228400,
        "weatherIconCode": 801,
      },
    ]
  }
  height={450}
  layout="horizontal"
  margin={
    Object {
      "bottom": 5,
      "left": 5,
      "right": 5,
      "top": 5,
    }
  }
  stackOffset="none"
  width={900}
>
  <Line
    activeDot={true}
    animationBegin={0}
    animationDuration={1500}
    animationEasing="ease"
    connectNulls={false}
    dataKey="temp"
    dot={<SummaryChartIcon />}
    fill="#fff"
    isAnimationActive={true}
    legendType="line"
    onAnimationEnd={[Function]}
    onAnimationStart={[Function]}
    points={Array []}
    stroke="#8884d8"
    strokeWidth={1}
    type="monotone"
    xAxisId={0}
    yAxisId={0}
  />
  <CartesianGrid
    fill="none"
    height={0}
    horizontal={true}
    horizontalPoints={Array []}
    stroke="#ccc"
    strokeDasharray="5 5"
    vertical={true}
    verticalPoints={Array []}
    width={0}
    x={0}
    y={0}
  />
  <XAxis
    allowDataOverflow={false}
    allowDecimals={true}
    dataKey="hour"
    domain={
      Array [
        0,
        "auto",
      ]
    }
    height={30}
    hide={false}
    mirror={false}
    orientation="bottom"
    padding={
      Object {
        "left": 0,
        "right": 0,
      }
    }
    reversed={false}
    scale="auto"
    tickCount={5}
    type="category"
    width={0}
    xAxisId={0}
  />
  <YAxis
    allowDataOverflow={false}
    allowDecimals={true}
    dataKey="temp"
    domain={
      Array [
        0,
        "auto",
      ]
    }
    height={0}
    hide={false}
    mirror={false}
    orientation="left"
    padding={
      Object {
        "bottom": 0,
        "top": 0,
      }
    }
    reversed={false}
    scale="auto"
    tickCount={5}
    type="number"
    width={60}
    yAxisId={0}
  />
  <Tooltip
    active={false}
    animationDuration={400}
    animationEasing="ease"
    content={<CustomTooltip />}
    coordinate={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    cursor={true}
    cursorStyle={Object {}}
    isAnimationActive={true}
    itemSorter={[Function]}
    itemStyle={Object {}}
    labelStyle={Object {}}
    offset={10}
    separator=" : "
    viewBox={
      Object {
        "x1": 0,
        "x2": 0,
        "y1": 0,
        "y2": 0,
      }
    }
    wrapperStyle={Object {}}
  />
</AnimationDecorator(ComposedDataDecorator(LineChart))>
`;
